cpg-library/src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/ScopeManager.java

```diff
@@ -563,8 +564,45 @@ public ValueDeclaration resolve(DeclaredReferenceExpression ref) {
     return resolve(currentScope, ref);
   }
 
+  /**
+   * Tries to resolve a function in a call expression.
+   *
+   * @param call the call expression
+   * @return a list of possible functions
+   */
   public List<FunctionDeclaration> resolveFunction(CallExpression call) {
-    return resolveFunction(currentScope, call);
+    var scope = currentScope;
+
+    // First, we need to check, whether we have some kind of scoping. this is currently only limited
```

Comment:
Do we want to have language specific code here?

Reply:
In the long run: No, that's why I created #499. I can do a quick check, whether this would destroy anything if I remove the language frontend guard. On the other hand, we do already have a language specific resolving for CXX.